src/collision/cm_model.c

```diff
@@ -471,6 +486,14 @@ cm_bsp_model_t *Cm_LoadBspModel(const char *name, int64_t *size) {
 	Cm_LoadBspAreas(&header.lumps[BSP_LUMP_AREAS]);
 	Cm_LoadBspAreaPortals(&header.lumps[BSP_LUMP_AREA_PORTALS]);
 
+	// unref the materials from the list
+	for (uint32_t i = 0; i < materials->len; ++i) {
+		cm_material_t *material = g_array_index(materials, cm_material_t *, i);
+		Cm_UnrefMaterial(material);
+	}
+
+	g_array_free(materials, true);
```

Comment:
What's the motivation for using `GArray` instead of an array of structs here? This is different from how we load everything else.